File: src/hashtable/LeetCode1189.py

from typing import Counter


class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        count = Counter(text)
        count['o'] >>= 1
        count['l'] >>= 1
        return min(count[c] for c in 'balon')


if __name__ == "__main__":
    text = 'loonbalxballpoon'
    res = Solution().maxNumberOfBalloons(text)
    print(res)

File: src/hashtable/LeetCode1189.java
package hashtable;//package hashtable;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class LeetCode1189 {
    public static void main(String[] args) {

        Solution solution = new Solution();
        String text = "loonbalxballpoon";
        int res = solution.maxNumberOfBalloons(text);
        System.out.println(res);
        
    }


    static class Solution{
        public int maxNumberOfBalloons(String text) {

            char [] chars = text.toCharArray();

            Set<Character> bSet = new HashSet<>(List.of('a', 'b','l','o','n'));

            int [] arr = new int[26];
            
            for(char c : chars){
                arr[c-'a'] += 1;
            }

            arr['l'-'a'] >>= 1;
            arr['o'-'a'] >>= 1;

            int res = Integer.MAX_VALUE;

            for(char c : bSet){
                res = Math.min(res, arr[c-'a']);
            }

            return res;
        }
    }
    
}
